### ISSUE TITLE
Wazuh Manager service not starting Error wazuh-authd: ERROR: Unable to connect to socket 'queue/db/wdb'

### ISSUE THREAD

**Issue by ketantande** [wazuh/wazuh#12320]
|Wazuh version|Component|Install type|Install method|Platform|
|WAZUH_VERSION":"v4.2.5"|Wazuh Manager Database|Distributed|Manual|Ubantu 20.04
| X.Y.Z-rev | Wazuh component | Manager/Agent | Packages/Sources | OS version |

2022/02/16 09:31:37 wazuh-authd: ERROR: Unable to connect to socket 'queue/db/wdb'.
2022/02/16 09:31:37 wazuh-authd: ERROR: Unable to connect to socket 'queue/db/wdb'
2022/02/16 09:31:37 wazuh-authd: ERROR: Error querying Wazuh DB to get the last agent keepalive.
2022/02/16 09:31:43 wazuh-authd: ERROR: Unable to connect to socket 'queue/db/wdb'.
2022/02/16 09:31:43 wazuh-authd: ERROR: Unable to connect to socket 'queue/db/wdb'.
2022/02/16 09:31:43 wazuh-authd: ERROR: Unable to connect to socket 'queue/db/wdb'
2022/02/16 09:31:43 wazuh-authd: ERROR: Error querying Wazuh DB to get the last agent keepalive.

there is no wdb file at /var/ossec/queue/db/

Please help me to resolve this issue.

**Comment by MiguelazoDS** [comment]
Hi @ketantande, 

Could you provide us with more information about what you were performing when this issue took place? 

As I can see from the logs you faced this issue when trying to replace an agent. Because of that keepalive error I see **authd** wanted to determine if the agent is old enough to be replaced according to the force_time setting in the **authd** configuration block.

But something else should have been caused the **wdb** to be deleted. Anyway, restarting Wazuh `wazuh-control restart` should be enough and the socket file should be re-generated.

Did you see some errors during a restart?  

Your agent is also Wazuh 4.2.5? 

If there's something wrong in your configuration file this could lead to not restarting properly your environment, and the wdb file won't be created.

**Comment by ketantande** [comment]
> Hi @ketantande,
> 
> Could you provide us with more information about what you were performing when this issue took place?
> 
> As I can see from the logs you faced this issue when trying to replace an agent. Because of that keepalive error I see **authd** wanted to determine if the agent is old enough to be replaced according to the force_time setting in the **authd** configuration block.
> 
> But something else should have been caused the **wdb** to be deleted. Anyway, restarting Wazuh `wazuh-control restart` should be enough and the socket file should be re-generated.
> 
> Did you see some errors during a restart?
> 
> Your agent is also Wazuh 4.2.5?
> 
> If there's something wrong in your configuration file this could lead to not restarting properly your environment, and the wdb file won't be created.

Hello MiguelazoDS,

Thanks for your reply

We were trying to forward wazuh logs to our SIEM server (We are only using Wazuh for FIM)
So We added  below strings in ossec.cong file and saved the changes and after that restarted Wazuh-manager server 
 from that time we are facing above error,so we have reverted all the changes but still we are getting reported error <ossec_config>
<syslog_output>
    <server>x.x.x.x</server>
  </syslog_output>
</ossec_config>

Ref : https://documentation.wazuh.com/current/user-manual/manager/manual-syslog-output.html 

I have tried to run wazuh-control commmand as advised by you and getting below result.

:/var/ossec/bin# sh wazuh-control restart
wazuh-clusterd not running...
wazuh-modulesd not running...
wazuh-monitord not running...
wazuh-logcollector not running...
wazuh-remoted not running...
wazuh-syscheckd not running...
wazuh-analysisd not running...
wazuh-maild not running...
wazuh-execd not running...
wazuh-db not running...
Killing wazuh-authd...
wazuh-agentlessd not running...
wazuh-integratord not running...
wazuh-dbd not running...
wazuh-csyslogd not running...
Killing wazuh-apid...
Wazuh v4.2.5 Stopped
Starting Wazuh v4.2.5...
Started wazuh-apid...
Started wazuh-csyslogd...
Started wazuh-dbd...
2022/02/17 04:38:53 wazuh-integratord: INFO: Remote integrations not configured. Clean exit.
Started wazuh-integratord...
Started wazuh-agentlessd...
Started wazuh-authd...
wazuh-db did not start correctly.

**Comment by MiguelazoDS** [comment]
Hi @ketantande, 

Thanks for the information provided. By taking a look at this output, I'm pretty sure we found the issue. 

I see that you have configured `wazuh-dbd`. The issue here is that when you configure `wazuh-dbd` the `wazuh-control` script wrongly assumes that `wazuh-db` has started due to the name similarity. 

This causes` wazuh-db` not to start as you showed here  `wazuh-db did not start correctly`. Ergo the wdb socket file is not created.

This bug was introduced as part of the daemon renaming in 4.2. `ossec-dbd` was renamed to `wazuh-dbd` and triggered this behavior. 

We already tracked this issue here

https://github.com/wazuh/wazuh/issues/10690

And also we implement a fix for this here.

https://github.com/wazuh/wazuh/pull/10849

But it is targeted to be released at Wazuh 4.3. And I'm afraid I don't have a date for that. 

Now I would like to ask you, are you really using this `wazuh-dbd` module?. Because is a legacy feature that is not being maintained. Disabling `wazuh-dbd` should be enough to start `wazuh-db` properly.

Regards!
